File: php/features.php
<?php
declare(strict_types=1);

// TempMailApi2 SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class TempMailApi2Features
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new TempMailApi2BaseFeature();
            case "test":
                return new TempMailApi2TestFeature();
            default:
                return new TempMailApi2BaseFeature();
        }
    }
}

File: py/features.py
# TempMailApi2 SDK feature factory

from feature.base_feature import TempMailApi2BaseFeature
from feature.test_feature import TempMailApi2TestFeature


def _make_feature(name):
    features = {
        "base": lambda: TempMailApi2BaseFeature(),
        "test": lambda: TempMailApi2TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
